compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -554,6 +554,7 @@ impl<'tcx> InferCtxt<'tcx> {
                     // We can't normalize associated types from `rustc_infer`,
                     // but we can eagerly register inference variables for them.
                     // FIXME(RPITIT): Don't replace RPITITs with inference vars.
+                    // FIXME(fmease): I'm pretty sure we need to do sth. similar for `ty::Inherent`.
```

Comment:
Yes, though similarly, not sure if the *old* (or new) solver is equipped to handle projection goals with IAT, so it might ICE.